compiler/src/utils/config.re

```diff
@@ -311,6 +337,22 @@ let principal =
     false,
   );
 
+let compilation_mode =
+  opt(
+    ~names=["compilation-mode"],
+    ~conv=
+      option_conv(
+        Cmdliner.Arg.enum([
+          ("normal", "normal"),
```

Comment:
Can cmdliner do something like this?
```suggestion
          ("normal", Env.normal),
```

Reply:
Yes, but `Env` depends on `Config`. We could potentially move it into some separate file that they can both depend on, though.